projects/portal/src/app/models/cosmos/staking.application.service.ts

```diff
@@ -270,6 +271,193 @@ export class StakingApplicationService {
+    gasRatio: number,
+    cosmosPublicKey: PubKey,
+    disableSimulate?: boolean,
+  ): Promise<InterfaceGasAndFee> {
+    if (!disableSimulate) {
+      const dialogRefSimulating = this.loadingDialog.open('Simulating...');
+
+      try {
+        const simulatedResultData = await this.staking.simulateToEditValidator(
+          editValidatorData,
+          minimumGasPrice,
+          cosmosPublicKey,
+          gasRatio,
+        );
+        const gas = simulatedResultData.estimatedGasUsedWithMargin;
+        const fee = simulatedResultData.estimatedFeeWithMargin;
+        return { gas, fee };
+      } catch (error) {
+        console.error(error);
+        const errorMessage = `Tx simulation failed: ${(error as Error).toString()}`;
+        this.snackBar.open(`An error has occur: ${errorMessage}`, 'Close');
+        const gas = { denom: 'uguu', amount: '1000000' };
+        const fee = null;
+        return { gas, fee };
+      } finally {
+        dialogRefSimulating.close();
+      }
+    }
+    const gas = { denom: 'uguu', amount: '1000000' };
+    const fee = null;
+    return { gas, fee };
+  }
+
+  async editValidatorSimple(
+    editValidatorData: EditValidatorData,
+    minimumGasPrice: proto.cosmos.base.v1beta1.ICoin,
+    privateKeyString: string,
+    gasRatio: number,
+    options?: InterfaceValidatorSimpleOptions,
+  ) {
+    const currentCosmosWallet = await this.walletService.currentCosmosWallet$
+      .pipe(take(1))
+      .toPromise();
+    if (!currentCosmosWallet) {
+      throw Error('Current connected wallet is invalid!');
+    }
+    const privateKey = convertHexStringToUint8Array(privateKeyString);
+    if (!privateKey) {
+      this.snackBar.open('Invalid PrivateKey!', 'Close');
+      return;
+    }
+
+    const dialogRef = this.loadingDialog.open('Sending Tx to be validator...');
+    const { disableRedirect, disableErrorSnackBar, disableSimulate } = options || {};
+
+    try {
+      const { gas, fee } = await this.handleSimulateToEditValidator(
+        editValidatorData,
+        minimumGasPrice,
+        gasRatio,
+        currentCosmosWallet.public_key,
+        disableSimulate,
+      );
+
+      const editValidatorResult = await this.staking.editValidator(
+        editValidatorData,
+        currentCosmosWallet,
+        gas,
+        fee,
+        privateKey,
+      );
+      const txHash = editValidatorResult.tx_response?.txhash;
+      if (txHash === undefined) {
+        throw Error('Invalid txHash!');
+      }
+      this.snackBar.open('Success', undefined, { duration: 6000 });
+      if (!disableRedirect) {
+        const redirectUrl =
+          location.port === '80' || location.port === '443' || location.port === ''
+            ? `${location.protocol}//${location.hostname}/explorer/validators/${editValidatorData.validator_address}`
+            : `${location.protocol}//${location.host}/explorer/validators/${editValidatorData.validator_address}`;
```

Comment:
This implementation was brought from createValidator.
Why do you redirect to the same port for ports 443 and 80?
I think `${location.protocol}//${location.host}/explorer/validators/${editValidatorData.validator_address};` is enough.

Reply:
I guess you are right. This is redundant and confusing. Can you fix it with your simple way? Thank you!